crates/erg_parser/error.rs

```diff
@@ -164,68 +219,18 @@ impl LexError {
         ))
     }
 
-    pub fn no_var_error(
-        errno: usize,
-        loc: Location,
-        name: &str,
-        similar_name: Option<String>,
-    ) -> Self {
-        let hint = similar_name.map(|n| {
-            let n = StyledString::new(n, Some(HINT), Some(Attribute::Bold));
-            switch_lang!(
-                "japanese" => format!("似た名前の変数があります: {n}"),
-                "simplified_chinese" => format!("存在相同名称变量: {n}"),
-                "traditional_chinese" => format!("存在相同名稱變量: {n}"),
-                "english" => format!("exists a similar name variable: {n}"),
-            )
-        });
-        let name = StyledString::new(name, Some(ERR), Some(Attribute::Underline));
-        Self::new(ErrorCore::new(
-            vec![SubMessage::ambiguous_new(loc, vec![], hint)],
-            switch_lang!(
-                "japanese" => format!("{name}という変数は定義されていません"),
-                "simplified_chinese" => format!("{name}未定义"),
-                "traditional_chinese" => format!("{name}未定義"),
-                "english" => format!("{name} is not defined"),
-            ),
+    pub fn invalid_convert_error(errno: usize, loc: Location, from: &str, to: &str) -> ParseError {
```

Comment:
The conversion from the RHS value when parsing the LHS value is an implementation detail that the user does not need to know.
It is better to simply make it a SyntaxError or indicate the token kind expected.

Reply:
I thought about converting this error to an error for the user to see, but decided not to include it in this PR
Therefore, I have reverted this error